src/AbstractPhpdocToTypeDeclarationFixer.php

```diff
@@ -51,7 +51,8 @@ abstract class AbstractPhpdocToTypeDeclarationFixer extends AbstractFixer implem
     use ConfigurableFixerTrait;
 
     private const REGEX_CLASS = '(?:\\\?+'.TypeExpression::REGEX_IDENTIFIER
-        .'(\\\\'.TypeExpression::REGEX_IDENTIFIER.')*+)';
+        .'(\\\\'.TypeExpression::REGEX_IDENTIFIER.')*+)'
+    ;
```

Comment:
👎 as splitting one line into 2 should not add 3rd line with semicolon in general. It only makes code longer and harder to write.

The same for ternary.

What is the general php CS and across other languages?

Reply:
The whole point of this rule option is to add an extra line for the semicolon. [It's always been like this.](https://cs.symfony.com/doc/rules/semicolon/multiline_whitespace_before_semicolons.html#example-2)

> splitting one line into 2 should not add 3rd line with semicolon in general. It only makes code longer and harder to write.

This is a subjective opinion. In the same way that I believe it makes code easier to read is subjective. Some might argue that this particuar line should be forced to be on a single line and never on multiple. All that said, the whole point of having configurable rules is so that you can choose what rules you want to use in your own projects. The CS fixer will "fix" it to follow those rules. Ultimately, the way you write it doesn't have to change. We shouldn't discourage nor prevent the inclusion of rules based on our own subjective opinions.

The reason this change is in this PR is because this library uses this rule already. CI requires you to run the fixer against the repo.

I'd also like to mention that the [braces rule](https://cs.symfony.com/doc/rules/basic/braces.html) does [exactly the same thing with curly braces](https://phptools.online/php-checker/result/54b844e5-4f41-4dbe-972b-e6ae15eea9dd#diff) when you split onto multiple lines.

Reply:
Chained calls are reasoned with minimal diff when adding/removing calls.

Ternary is however definitely ternary, hence the name. Splitting a long line with an operator does not make sense to me either. Yes, this is my opinion. AFAIK, C# or AirBnB JS CS keep the semicolon on the same line too.

Reply:
I want to save real estate (LoC) and have this CS confirmed by other major coding standards. So far I see no major coding standards behave like this.

Reply:
Then don't use this configuration. This is an optional feature.

Reply:
So it should not be default even for this project. It should be under opt-in option. Not mentioning a CS break for other projects.

Reply:
It's not a default. It's included in the [PhpCsFixer](https://cs.symfony.com/doc/ruleSets/PhpCsFixer.html) rule set because PhpCsFixer repos already use this rule. It's expected output from this rule. As mentioned before, [this rule has always added extra lines for semicolons](https://cs.symfony.com/doc/rules/semicolon/multiline_whitespace_before_semicolons.html#example-2).

If you want to be following a industry standardised ruleset you should probably be using PER-CS / PSR-12 rule set. That does not include this rule.

Reply:
A ternary expression split across multiple lines _is_ a multiline chained call. The statements are between meaningful operator tokens. Thus, chained.

| statement | operator | statement | operator | statement |
|:-:|:-:|:-:|:-:|:-:|
| `isset($foo)` | `?` | `'a'` | `:` | `'b'` |

Reply:
Ok, we both understand the "chained call" differently.

https://en.wikipedia.org/wiki/Method_chaining

My definition is "call result needs to be used again", like `$a->a()->b()`.

Ternary is branching between left/right side based on a condition, it has no relation to "chained call", ternary does not even has to call any method (`$a ? $b : $c`), thus not related to `multiline_whitespace_before_semicolons` rule.

Reply:
If it has a semicolon then it's related to `multiline_whitespace_before_semicolons`. So ternary is related.

Method chaining is a form of chaining, yes. There are other things that can be "chained", though. Such as concatination or arithmetic. Examples:

```php
$foo = 1
    + 2
    * 3
;

$bar = 'this'
    . 'is'
    . 'chained'
    . 'concatenation'
;
```

It would be reasonable to expect the semicolon placement to be on the new line for the above when using the `new_line_for_chained_calls` option. If it didn't then it would be odd and inconsistent code styling (which is what currently happens before this PR):

```php
$foo = 1
    + 2
    * 3;

$bar->foo()
    ->bar()
;
```

`+`, `*`, `.` and indeed `?` and `:` are all operators chained inbetween a `semicolon` thus should all result in the semicolon placed on a new line.

As per my previous examples with multiple ternary statments they should also have the semicolon placed on a new line otherwise it's all inconsistent.